### DISCUSSION TITLE
Got rate limited error when trying to create new project for second time

### DISCUSSION THREAD

**Discussion by Senbonzakura1234** [solidjs/solid-start#564] [General]
**Step to reproduce:**

- create a new solid project once.
- create a new solid project again.

**Expected behavior:**    create both project successfully.

**Actual behavior:**      can't create 2nd project.

**Error screenshot:**

![image](https://user-images.githubusercontent.com/43206347/200985239-211dfb1a-baf5-4889-934e-025de4cfa89d.png)

**Comment by boehs** [comment]
This is a github ratelimit, I don't think we can do anything about it https://github.com/solidjs/solid-start/blob/d3318b2972f5f1323e0a945e49bc1f2bb51d426a/packages/create-solid/cli/github.js#L14-L25
I will say, this won't trigger after only two. Likely you have some other program connecting to github API as well like extension

**Comment by Senbonzakura1234** [comment]
> This is a github ratelimit, I don't think we can do anything about it
> 
> https://github.com/solidjs/solid-start/blob/d3318b2972f5f1323e0a945e49bc1f2bb51d426a/packages/create-solid/cli/github.js#L14-L25
> 
> 
> I will say, this won't trigger after only two. Likely you have some other program connecting to github API as well like extension

strange, i'm using cmd right now, not even open vscode yet.

**Comment by boehs** [comment]
I was more talking about chrome extensions. Regardless, [the API docs](https://docs.github.com/en/rest/overview/resources-in-the-rest-api) says you may make up to 60 API requests per hour without authentication. I don't know how many requests each starter makes, and what counts as an API request, but I'm pretty sure it's nowhere close to 31+.

THAT BEING SAID, if more people report this, we can look into delivering starters from our CDN, netlify

**Comment by Senbonzakura1234** [comment]
> I was more talking about chrome extensions. Regardless, [the API docs](https://docs.github.com/en/rest/overview/resources-in-the-rest-api) says you may make up to 60 API requests per hour without authentication. I don't know how many requests each starter makes, and what counts as an API request, but I'm pretty sure it's nowhere close to 31+.
> 
> THAT BEING SAID, if more people report this, we can look into delivering starters from our CDN, netlify

I don't think the problem caused by my chrome extension. But I noticed 1 thing that I only got rate limited when I'm using my company's network to run the `init` command. This doesn't happened when I tried it at home or using 4G. Is there any chance that my company's network did something to cause this?

**Comment by pikhosh** [comment]
I got the same error on the first run and after almost an hour I was able to use the `npm init solid@latest` command...

**Comment by dario-piotrowicz** [comment]
I've been hitting this issue a bunch as I am usually under a VPN so I very very often get rate limited by create-solid (I was about to open a new issue but then I checked and found this discussion), basically almost every time I try to create a new solid application I need to disable my VPN.

I also think that this could be a common issue when a simple IP is shared across different machines, such as what can happen with VPNs, conferences or shared office spaces.

So if possible I would really suggest not using the github rest API (which introduces this rate limiting), one very simple solution that I can think of is generating the examples list at build time and not having to make any request at all for it.

<details>
<summary>Also...</summary>

I've been poking at the code when looking into this, it seems like the create-solid package does sort of try to include a github token to maybe avoid the rate limiting:
![Screenshot 2024-01-28 at 16 22 37](https://github.com/solidjs/solid-start/assets/61631103/86a9ffc0-4f2e-4216-a3dd-fc441d36d5b8)
([source](https://github.com/solidjs/solid-start/blob/5d76739861803619f1d180230e0c8b16cb42eb23/packages/create-solid/cli/github.js#L25))

However it looks to me that there is no code that actually retrieves the token:
![Screenshot 2024-01-28 at 16 27 07](https://github.com/solidjs/solid-start/assets/61631103/44dc96f8-f912-4ef3-b896-bedb53e4ca2c)
([source](https://github.com/solidjs/solid-start/blob/5d76739861803619f1d180230e0c8b16cb42eb23/packages/create-solid/cli/index.js#L101-L109))

So this feels a bit off to me 😕 🤷  (maybe there was a plan at some point to get the token from somewhere?)

</details>
